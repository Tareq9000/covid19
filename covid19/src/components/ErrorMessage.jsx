import React, { Component } from 'react';
import ErrorIcon from '@material-ui/icons/Error';
import { connect } from 'react-redux';
import styles from '../styles/ErrorMessage.module.css';
import { getGlobalSummary } from '../reducers/covidReducer.js';

export class ErrorMessage extends Component{

    render() {
        const { tryAgain } = this.props
      return (
        <div className={styles.box}>
            <div className={styles.innerBox$}>
                <ErrorIcon className={styles.icon}/>
                <div className={styles.text}>ERROR! It seems like there is a Problem with getting the data. Please try agin</div>
                <button onClick={tryAgain}>Try again</button>
            </div>
        </div>
      );
    }
  }

  const mapDispatchToProps = ( dispatch ) => {
 
    return {
        tryAgain: () => (
            dispatch(getGlobalSummary())
        )
    }
}
export default connect(null, mapDispatchToProps)(ErrorMessage)