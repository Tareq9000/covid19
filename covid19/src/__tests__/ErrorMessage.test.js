import { ErrorMessage  } from '../components/ErrorMessage.jsx';

import React from 'react';
import { shallow } from 'enzyme';

describe('ErrorMessage test', () => {

    it('should render correctly', () => {
        const component = shallow(<ErrorMessage />);
      
        expect(component).toMatchSnapshot();
    });

    it('should refetch by pressing the button', () => {
        const func = jest.fn()
        const component = shallow(<ErrorMessage tryAgain={func} />);
      
        component
            .find('button')
            .simulate('click');

        expect(func).toHaveBeenCalled();

    });
})