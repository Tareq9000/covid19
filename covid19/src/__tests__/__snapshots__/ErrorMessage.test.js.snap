// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage test should render correctly 1`] = `
<div
  className="box"
>
  <div
    className="innerBox$"
  >
    <ErrorIcon
      className="icon"
    />
    <div
      className="text"
    >
      ERROR! It seems like there is a Problem with getting the data. Please try agin
    </div>
    <button>
      Try again
    </button>
  </div>
</div>
`;
